staging/kos/pkg/registry/network/subnet/strategy.go

```diff
@@ -167,15 +137,13 @@ func (ss *subnetStrategy) checkNSAndCIDRConflicts(candidate *subnet.Summary, mal
 		return
 	}
 	glog.V(5).Infof("Found %d subnets with vni %d", len(potentialRivals), candidate.VNI)
+	// Check whether there are Namespace and CIDR conflicts with other subnets.
 	for _, potentialRival := range potentialRivals {
-		pr, err := subnet.NewSummary(potentialRival)
-		if err != nil || candidate.SameSubnetAs(pr) {
```

Comment:
No need to check whether `potentialRival` is the same subnet as the one under validation and skip it if that's the case because after immutability this code is only executed upon subnet creation.